src/device-config.ts

```diff
@@ -395,10 +397,39 @@ export class DeviceConfig {
 						target[envVarName],
 					)
 				) {
-					// Save the change if it is
-					configChanges[key] = target[envVarName];
-					humanReadableConfigChanges[envVarName] = target[envVarName];
-					reboot = rebootRequired || reboot;
+					// Check that the difference is not due to the variable having an invalid
+					// value set from the cloud
+					if (
+						this.config.valueIsValid(key as SchemaTypeKey, target[envVarName])
+					) {
+						// Save the change if it is both valid and different
+						changingValue = target[envVarName];
+					} else {
+						const message = `Warning: Ignoring invalid device configuration value for ${key}, value: ${inspect(
+							target[envVarName],
+						)}. Falling back to default (${defaultValue})`;
+						this.logger.logSystemMessage(
+							message,
+							{ key: envVarName, value: target[envVarName] },
+							'invalidDeviceConfig',
+							false,
+						);
+						if (
+							!DeviceConfig.configTest(
+								varType,
+								current[envVarName],
+								defaultValue,
+							)
+						) {
+							// Set it to the default value if it is different to the current
```

Comment:
If you move the warning log here, you can make it be logged only once, otherwise it'll be logged on every call to getRequiredSteps (because current and target will never be the same, but current and default will)

Reply:
Ah yeah, very good point - only report it when we initially fall back to the default :+1: